bsc/header.go

```diff
@@ -185,15 +184,22 @@ func Keccak256(data ...[]byte) []byte {
 }
 
 // SealHash returns the hash of a block prior to it being sealed.
-func SealHash(header *Header) (hash Hash) {
+func SealHash(header *Header, chainID *big.Int) (hash Hash) {
 	hasher := sha3.NewLegacyKeccak256()
-	encodeSigHeader(hasher, header)
+	encodeSigHeader(hasher, header, chainID)
 	hasher.Sum(hash[:0])
 	return hash
 }
 
-func encodeSigHeader(w io.Writer, header *Header) {
-	err := rlp.Encode(w, []interface{}{
+func encodeSigHeader(w io.Writer, header *Header, chainId *big.Int) {
+	var err error
+	var content []interface{}
+
+	if chainId != nil {
+		content = append(content, chainId)
+	}
+	content = append(content, []interface{}{
+		chainId,
```

Comment:
remove this line, otherwise `chainId` will append twice if it's not `nil`

Reply:
ok